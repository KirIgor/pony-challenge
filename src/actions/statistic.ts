import { ActionCreator } from 'redux';

// action types
export const ADD_WINS = 'ADD_WINS';
export type ADD_WINS = typeof ADD_WINS;

export const ADD_LOSES = 'ADD_LOSES';
export type ADD_LOSES = typeof ADD_LOSES;

// actions

export interface AddWins {
	type: ADD_WINS;
}

export interface AddLoses {
	type: ADD_LOSES;
}

export type StatisticAction = AddWins | AddLoses;

// action creators

export const addWins: ActionCreator<AddWins> = () => ({ type: ADD_WINS });
export const addLoses: ActionCreator<AddLoses> = () => ({ type: ADD_LOSES });
