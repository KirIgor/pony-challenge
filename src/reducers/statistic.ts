import { StatistcState } from '../types/index';
import { StatisticAction, ADD_WINS, ADD_LOSES } from '../actions/statistic';

export const defaultStatisticState = new StatistcState({
	wins: 0,
	loses: 0
});

const statistcReducer = (
	state: StatistcState = defaultStatisticState,
	action: StatisticAction
): StatistcState => {
	switch (action.type) {
		case ADD_WINS: {
			return state.update('wins', wins => wins + 1) as StatistcState;
		}
		case ADD_LOSES: {
			return state.update('loses', loses => loses + 1) as StatistcState;
		}
		default:
			return state;
	}
};

export default statistcReducer;
