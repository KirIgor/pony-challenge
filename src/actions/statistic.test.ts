import { addWins, addLoses } from './statistic';

describe('statistic action creators', () => {
	describe('addWins action creator', () => {
		it('should dispatch correct action', () => {
			expect(addWins()).toMatchSnapshot();
		});
	});

	describe('addLoses action creator', () => {
		it('should dispatch correct action', () => {
			expect(addLoses()).toMatchSnapshot();
		});
	});
});
