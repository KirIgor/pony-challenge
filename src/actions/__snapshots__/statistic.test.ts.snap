// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`statistic action creators addLoses action creator should dispatch correct action 1`] = `
Object {
  "type": "ADD_LOSES",
}
`;

exports[`statistic action creators addWins action creator should dispatch correct action 1`] = `
Object {
  "type": "ADD_WINS",
}
`;
